swap/src/bin/asb.rs

```diff
@@ -377,3 +373,24 @@ async fn register_tor_services(
 
     Ok(ac)
 }
+
+fn get_env_config_from_defaults_and_config(testnet: bool, config: &Config) -> env::Config {
+    let env_config = if testnet {
+        env::Testnet::get_config()
+    } else {
+        env::Mainnet::get_config()
+    };
+
+    let env_config =
+        if let Some(bitcoin_finality_confirmations) = config.bitcoin.finality_confirmations {
+            env_config.with_bitcoin_finality_confirmations(bitcoin_finality_confirmations)
+        } else {
+            env_config
+        };
+
+    if let Some(monero_finality_confirmations) = config.monero.finality_confirmations {
+        env_config.with_monero_finality_confirmations(monero_finality_confirmations)
+    } else {
+        env_config
+    }
+}
```

Comment:
This looks like it would like to live within the `env` module so you can say:

```rust
let env_config = env::new(testnet, config);
```

Reply:
The motivation should always be to have as little code inside `main.rs` as possible.

Reply:
I did not do that because `env` is share between CLI and ASB and the config is only relevant for the ASB. I can move it there, but I don't find that especially clean neither.

Reply:
Fair point, I didn't think of that.